src/lib/hooks/useUserData/index.ts

```diff
@@ -44,132 +32,91 @@ const fetchUser: UserFetcherSignature = async (userId, isLoadingAuth) => {
   return user;
 };
 
-const fetchProjects: ProjectsFetcherSignature = async (
+type SensorsFetcherSignature = (
+  userId?: AuthenticatedUsersType["id"],
+  isLoadingAuth?: boolean
+) => Promise<definitions["sensors"][] | null>;
+
+const fetchUserSensors: SensorsFetcherSignature = async (
   userId,
   isLoadingAuth
 ) => {
   if (isLoadingAuth || isLoadingAuth === undefined) return null;
   if (!userId) throw new Error("Not authenticated");
 
   const { data, error } = await supabase
-    .from<ProjectsType>("projects")
-    .select(
-      `
-      id,
-      name,
-      description,
-      connectype,
-      location,
-      category: categoryId (
-        id,
-        name,
-        description
-      ),
-      devices (
-        id,
-        externalId,
-        name,
-        records (
-          id,
-          recordedAt
-        )
-      )
-    `
-    )
-    .eq("userId", userId);
+    .from<definitions["sensors"]>("sensors")
+    .select(sensorQueryString)
+    .eq("user_id", userId);
 
   if (error) throw error;
   else if (!data)
-    throw new Error(`Projects for user with id "${userId} was not found"`);
+    throw new Error(`Sensors for user with id "${userId} not found"`);
   return data;
 };
 
-const addDevice = async (device: Omit<DevicesType, "id">): Promise<void> => {
-  if (!device.userId) throw new Error("Not authenticated");
-
-  const { error } = await supabase
-    .from<DevicesType>("devices")
-    .insert([{ ...device, id: undefined }]);
-
-  if (error) throw error;
-};
-
-const updateDevice = async ({
-  id,
-  userId,
-  externalId,
-  name,
-}: Partial<DevicesType>): Promise<void> => {
-  if (!userId) throw new Error("Not authenticated");
-
-  const { error } = await supabase
-    .from<DevicesType>("devices")
-    .update({ externalId, name })
-    .eq("id", id)
-    .eq("userId", userId);
-
-  if (error) throw error;
-};
-
-const deleteDevice = async (
-  id: DevicesType["id"],
-  userId: string | undefined
+const createSensor = async (
+  sensor: Omit<definitions["sensors"], "id">
 ): Promise<void> => {
-  if (!userId) throw new Error("Not authenticated");
+  if (!sensor.user_id) throw new Error("Not authenticated");
```

Comment:
Here it isn't about the sensor's `user_id` but it's about verifying that the user is authenticated. This should come from `useAuth` or similar. It represents the id of the logged-in user.

Reply:
My bad. In the diff, this looks as if you're deleting the sensor but you are creating one. Ignore this

Reply:
You're actually right, we should get the `user_id` from `useAuth`. I changed that now in 6f72974.